freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/PasswordService.java

```diff
@@ -121,6 +130,25 @@ private void asyncSetPasswords(Optional<String> requestId, String operationId,
         }
     }
 
+    @VisibleForTesting
+    Optional<Instant> calculateExpirationTime(String actorCrn, String accountId) {
+        LOGGER.debug("calculating expiration time for password in account {}", accountId);
+        UserManagementProto.Account account = umsClient.getAccountDetails(actorCrn, accountId, MDCUtils.getRequestId());
```

Comment:
i had one nagging thought about whether we should use the internal actor here rather than the caller. i don't know whether the getAccount call would need rights that users may not have.

Reply:
So there is no authorization for this in the backend, but I think that calling this using the internal actor is better as it is future proofing this in case we change the authorization on the get-account call in the future. That said, feel free to do it in a follow-up commit.